lib/kms_monitor/cloudtrail.rb

```diff
@@ -0,0 +1,159 @@
+require 'logger'
+require 'aws-sdk-dynamodb'
+
+module IdentityKMSMonitor
+
+  # Class for inputting CloudTrail events into DynamoDB
+  class CloudTrailToDynamoHandler
+
+    attr_reader :dynamo
+
+    def initialize(log_level: Logger::INFO, dry_run: true, dynamo: nil)
+      log.level = log_level
+      log.debug("Initializing, dry_run: #{dry_run.inspect}")
+
+      @dry_run = dry_run
+
+      begin
+        @dynamo = dynamo || Aws::DynamoDB::Client.new
+      rescue StandardError
+        log.error('Failed to create DynamoDB client. Do you have AWS creds?')
+        raise
+      end
+    end
+
+    # @return [String]
+    def main(_args)
+      event = JSON.parse(STDIN.read)
+      inner_process(event)
+    end
+
+    # @lambda_event and @lambda_context are instance variables we inherit from the parent
```

Comment:
These instances variables actually aren't inherited. This class doesn't inherit from `AbstractLambdaHandler`, and you can't inherit instance variables anyway.

I finally got around to taking a stab at the refactor of the API for handlers here. In the new model, your handler class should directly subclass `Functions::AbstractLambdaHandler`, and then it defines a method to directly accept the event/context for execution: `def lamda_main(event:, context:)`.

This got pretty large, so I put it into a separate PR here for merging into this branch: https://github.com/18F/identity-lambda-functions/pull/2

What do you think?